js/body_mixin.ts

```diff
@@ -0,0 +1,230 @@
+    }
+    return this.stream;
+  }
+
+  get isStatic(): boolean {
+    return (
+      typeof this.bodySource === "string" ||
+      this.bodySource instanceof Uint8Array
+    );
+  }
+
+  get staticBody(): Uint8Array {
+    if (this.bodySource instanceof Uint8Array) return this.bodySource;
+    else if (typeof this.bodySource === "string")
+      return new TextEncoder().encode(this.bodySource);
+    else throw new TypeError("body is not static");
+  }
+
+  set body(value: ReadableStream) {
+    this.stream = value;
+  }
+
+  get bodyUsed(): boolean {
+    if (this.body && this.body.locked) {
+      return true;
+    }
+    return false;
+  }
+
+  async blob(): Promise<Blob> {
+    return new DenoBlob([await this.arrayBuffer()]);
+  }
+
+  async formData(): Promise<FormData> {
+    if (this.bodySource instanceof DenoFormData) {
+      return this.bodySource;
+    }
+
+    const raw = await this.text();
+    const query = queryParse(raw);
+    const formdata = new DenoFormData();
+    for (let key in query) {
+      const value = query[key];
+      if (Array.isArray(value)) {
+        for (let val of value) {
+          formdata.append(key, val);
+        }
+      } else {
+        formdata.append(key, String(value));
+      }
+    }
+    return formdata;
+  }
+
+  async text(): Promise<string> {
+    if (typeof this.bodySource === "string") {
+      return this.bodySource;
+    }
+
+    const arr = await this.arrayBuffer();
+    return new TextDecoder("utf-8").decode(arr);
+  }
+
+  async json(): Promise<any> {
+    const raw = await this.text();
+    return JSON.parse(raw);
+  }
+
+  async arrayBuffer(): Promise<ArrayBuffer> {
+    if (
+      this.bodySource instanceof Int8Array ||
+      this.bodySource instanceof Int16Array ||
+      this.bodySource instanceof Int32Array ||
+      this.bodySource instanceof Uint8Array ||
+      this.bodySource instanceof Uint16Array ||
+      this.bodySource instanceof Uint32Array ||
+      this.bodySource instanceof Uint8ClampedArray ||
+      this.bodySource instanceof Float32Array ||
+      this.bodySource instanceof Float64Array
+    ) {
+      return <ArrayBuffer>this.bodySource.buffer;
+    } else if (this.bodySource instanceof ArrayBuffer) {
+      return this.bodySource;
+    } else if (typeof this.bodySource === "string") {
+      const enc = new TextEncoder();
+      return <ArrayBuffer>enc.encode(this.bodySource).buffer;
+    } else if (this.bodySource instanceof ReadableStream) {
+      return bufferFromStream((this.bodySource as ReadableStream).getReader());
+    } else if (this.bodySource instanceof DenoFormData) {
+      const enc = new TextEncoder();
+      return <ArrayBuffer>enc.encode(this.bodySource.toString()).buffer;
+    } else if (!this.bodySource) {
+      return new ArrayBuffer(0);
+    }
+    throw new Error(
+      `Body type not yet implemented: ${this.bodySource.constructor.name}`
+    );
+  }
+}
+
+/** @hidden */
+function validateBodyType(owner: any, bodySource: any) {
```

Comment:
tslint won't allow this `any`, I don't this has been run through the linter.  Can't we be more specific here.